### ISSUE TITLE
adapt actuated tls to work with meso

### ISSUE THREAD

**Issue by namdre** [eclipse-sumo/sumo#8735]
based on the predicted headways at the stop line instead of with induction loops.

**Comment by ambuehll** [comment]
I hope you don’t mind me following up here. Given the progress made in https://github.com/eclipse-sumo/sumo/pull/17191, would this now be feasible to implement with limited effort?

**Comment by namdre** [comment]
Not quite, because actuated trafflic lights currently rely on microsopic induction loops (and mesoscopic loops are not suitable). It's fixable with moderate effort but #17191 doesn't really help.

**Comment by ambuehll** [comment]
OK, thanks a lot! that would be amazing.

**Comment by namdre** [comment]
Implementation notes: 
- meso notifies detector whenever a vehicle leaves a segment. This is sufficient to establish headways for gap control but doesn't allow detecting vehicle presence for optional (on-request) phases.
- stopped vehicles must be detected by looking at the first-on-segment-vehicle waitingTime.
